src/main/java/org/spectrumauctions/sats/core/model/bvm/BMWorld.java

```diff
@@ -72,7 +73,7 @@ public BMBand getBand(String bandName) {
         for (BMBand band : bands) {
             licenses.addAll(band.getLicenses());
         }
-        return licenses;
+        return ImmutableSet.copyOf(licenses);
```

Comment:
Two remarks: 
- There is no need to make the Set immutable, as we are not referring an internal field of our class (i.e., we are not risking that an external user modifies the world by modifying the returned collection).
- If you want to return an immutable collection, it might be preferable to return _Collections.unmodifiableSet(licenses)_ instead, which is slightly more performant and, in this context is has no disadvantages over the Guava-ImmutableSet approach

Reply:
Good points, thanks!